test/core/test_promise.c

```diff
@@ -334,13 +334,131 @@ test_all_settled(void** result, void* data, void* value) {
   emscripten_promise_destroy(null_in[2]);
 
   em_promise_t to_finish[3] = {empty_checked, full_checked, null_checked};
-  em_promise_t finish_test_all = emscripten_promise_all(to_finish, NULL, 3);
+  em_promise_t finish_test_all_settled =
+    emscripten_promise_all(to_finish, NULL, 3);
 
   emscripten_promise_destroy(empty_checked);
   emscripten_promise_destroy(full_checked);
   emscripten_promise_destroy(null_checked);
 
-  *result = finish_test_all;
+  *result = finish_test_all_settled;
+  return EM_PROMISE_MATCH_RELEASE;
+}
+
+typedef struct promise_any_state {
+  size_t size;
+  em_promise_t in[3];
```

Comment:
What are these `3` constants?

Reply:
These are the (up to) 3 input promises. They're not actually needed after the initial call to `emscripten_promise_any`, but it's convenient to bundle all the input, output, and expected state together in a single struct.

Reply:
I'm confused, I don't see any mention of a limit of 3 on input promises in `promise.h`? The methods all take a `num_promises` without a mention of a limit in the comments.

Reply:
Oh ok, sorry, that's what I was missing...

Reply:
Yeah, this C API is a little funky because 100% of the implementation is in JS. There's no .c implementation file for the API at all.